lms/djangoapps/grades/signals/handlers.py

```diff
@@ -179,6 +179,11 @@ def enqueue_subsection_update(sender, **kwargs):  # pylint: disable=unused-argum
     enqueueing a subsection update operation to occur asynchronously.
     """
     _emit_problem_submitted_event(kwargs)
+
+    if kwargs['score_db_table'] == ScoreDatabaseTableEnum.submissions and kwargs['weighted_possible'] == 0:
```

Comment:
When I suggested handlers.py, I was thinking of updating `submissions_score_set_handler` so the submissions table logic remains within the handler for that database.